scripts/vsc-venv-wrapper/vsc-venv.sh

```diff
@@ -0,0 +1,252 @@
+
+  # Warn user if system python is used
+  if [ "$(which python)" = "/usr/bin/python" ]; then
+    echo_warning "System python used. Consider loading a specific Python module through the modules file."
+  fi
+
+  # Check if venv already exists
+  if [ -f "$venv_location/bin/activate" ]; then
+    # The virtual environment already exists
+    echo_info "Virtual environment already exists at $venv_location"
+
+    # Warn user if a different python module is loaded than the one used to create the virtual environment
+    local current_python venv_python
+    current_python=$(realpath "$(which python)")
+    venv_python=$(realpath -m "$venv_location/bin/python")
+
+    if [ "$current_python" != "$venv_python" ]; then
+      echo_error "The python module used to create the virtual environment is different from the one currently loaded."
+      echo_error "Current python module:  $current_python"
+      echo_error "venv python module:     $venv_python"
+      echo_error "If you changed the python module in $modules_file,"
+      echo_error "please remove the virtual environment that uses the old python with the following command:"
+      echo_error "  $ rm -rf $venv_location"
+      echo_error "After removing the virtual environment, run the activate command again."
+      echo_error "Exiting..."
+      return 1
+    fi
+
+  else
+    # The virtual environment does not exist
+    echo_info "Virtual environment does not exist at $venv_location"
+    python_version=$(python --version)
+    echo_info "Using $python_version to create virtual environment at $venv_location"
+
+    # Will automatically make the venvs folder and venv, does nothing if they already exist
+    if ! python -m venv "$venv_location"; then
+      echo_error "Could not create virtual environment"
+      return 1
+    fi
+  fi
+
+  # === Activate Virtual Environment === #
+
+  echo_info "Activating virtual environment"
+  source "$venv_location/bin/activate"
+
+  # === Warn user if requirements contain lines without version specifiers === #
+  local version_specifiers="==|>=|<=|!=|<|>|~="
+  local lines_without_versions=$(grep -vE "$version_specifiers" "$requirements_file")
+
+  # Check if the variable is not empty
+  if [ -n "$lines_without_versions" ]; then
+      echo_warning "The following lines do not contain version specifiers:"
+      echo_warning "$lines_without_versions"
+  fi
+
+
+  # === Install Requirements === #
+
+  echo_info "Installing requirements from '$requirements_file'"
+  if ! pip install -r "$requirements_file"; then # This will finish quickly if the requirements are already installed
+    echo_error "Could not install requirements"
+    deactivate
+    return 1
+  fi
+
+  echo_info "Requirements installed successfully"
+  echo_info "You can now run your Python scripts"
+}
+
+deactivate_() {
+  echo_info "Deactivating virtual environment at $VIRTUAL_ENV"
+  deactivate # For now, just use the python `deactivate`
```

Comment:
Also run `module purge` here?

Reply:
done in [purge modules when deactivating](https://github.com/hpcugent/vsc_user_docs/pull/670/commits/1d896faeee03f0f0ed02b5a68c88a4571e88dbf1)